cmd/dp-deployer/main.go

```diff
@@ -128,38 +137,33 @@ func initHandlers(ctx context.Context, cfg *config.Configuration, vc *vault.Clie
 	}, nil
 }
 
-func startHealthChecks(ctx context.Context, cfg *config.Configuration, vaultChecker *vault.Client, s3sChecker *s3client.S3, s3dChecker *s3client.S3) *healthcheck.HealthCheck {
-	hasErrors := false
+func startHealthChecks(ctx context.Context, cfg *config.Configuration, vaultChecker *vault.Client, s3sChecker *s3client.S3, s3dChecker *s3client.S3, nomadHealthClient *health.Client) (*healthcheck.HealthCheck, error) {
 
 	// Create healthcheck object with versionInfo
 	versionInfo, err := healthcheck.NewVersionInfo(BuildTime, GitCommit, Version)
 	if err != nil {
-		log.Event(ctx, "failed to create service version information", log.FATAL, log.Error(err))
-		os.Exit(1)
+		return nil, errors.Wrap(err, "failed to create service version information")
 	}
 	hc := healthcheck.New(versionInfo, cfg.HealthcheckCriticalTimeout, cfg.HealthcheckInterval)
 
 	if err := hc.AddCheck("Vault", vaultChecker.Checker); err != nil {
-		hasErrors = true
```

Comment:
I would have kept `hasErrors` (maybe rename to `hasCheckErrors` but kept functionality), to capture all checks that failed by logging each failed check before returning.

Reply:
After re-thinking the problem - it does add a level of complexity to the code just so we gain more via our logs for debugging if something went wrong. Im happy for you to stick with Jess' suggestion